src/convertStringToValue.ts

```diff
@@ -0,0 +1,70 @@
+import { EventPayload } from './EventPayload';
+
+// Define the list of possibly numerical properties to check and convert
+const propertiesToCheck = [
```

Comment:
I looked into this a little bit and it appears that since interfaces do not exist at runtime, extracting any information from it at run time is not possible. It seems in order to use `typeof` on fields of EventPayload the object would need to be initialized. 
I also considered just looping through the data object and just determining to convert based on if it is a string representing a boolean or a number, but I am concerned that this will cause more complexity for fields that are strings that look like numbers, but are meant to stay strings in the payload. For example, browserVersion and alike fields. 

I know its not ideal to maintain a hard coded list, but I can add this to the wiki for something to remember when adding a new field to the events schema.

Reply:
gotcha thanks for the explanation.